src/assembler.jl

```diff
@@ -208,3 +208,23 @@ function InsertionSort!(A, order, ii=1, jj=length(A))
     end  # i
     return
 end
+
+"""
+    disassemble!(ue, u, dofs)
+
+Fill the element vector `ue` with values from the global vector `u`.
+
+This function is equivalent to `ue .= u[dofs]` but without allocating
+the intermediate array.
+"""
+@propagate_inbounds function disassemble!(ue::AbstractVector, u::AbstractVector, dofs::AbstractVector{Int})
+    @boundscheck begin
+        length(ue) == length(dofs) || throw(BoundsError())
```

Comment:
I don't think this is worth it. It will only prevent work in the case that the code will error and in that case, the performance doesn't really matter.